README.md

```diff
@@ -254,8 +254,135 @@ class MysqlSpec extends FlatSpec with ForAllTestContainer {
 }
 ```
 
+## New API
+
+Starting from 0.34.0 version testcontainers-scala provides the new API. 
+The main motivation points are in the [pull request](https://github.com/testcontainers/testcontainers-scala/pull/78).
+
+### `Container` and `ContainerDef` 
+
+Docker containers are represented through the two different entities:
+1. `ContainerDef` — it's container definition. `ContainerDef` describes, how to build a container.
+   You can think about it like about a container constructor, or dockerfile description.
+   Usually, `ContainerDef` receives some parameters.
+   `ContainerDef` has a `start()` method. It returns a started `Container`.
+2. `Container` — it's a started container. You can interact with it through its methods.
+   For example, in the case of `MySQLContainer` you can get it's JDBC URL with `jdbcUrl` method.
+   `Container` is the main entity for using inside tests.
+
+### Scalatest usage
+
+You can use one of the four traits:
+1. `TestContainerForAll` — will start a single container before all tests and stop after all tests.
+2. `TestContainerForEach` — will start a single container before each test and stop after each test.
+3. `TestContainersForAll` — will start multiple containers before all tests and stop after all tests.
+4. `TestContainersForEach` — will start multiple containers before each test and stop after each test.
+
+
+#### Single container in tests
+
+If you want to use a single container in your test:
+```scala
+class MysqlSpec extends FlatSpec with TestContainerForAll {
+
+  // You need to override `containerDef` with needed container definition
+  override val containerDef = MySQLContainer.Def()
+
+  // To use containers in tests you need to use `withContainers` function
+  it should "test" in withContainers { mysqlContainer =>
+    // Inside your test body you can do with your container whatever you want to
+    assert(mysqlContainer.jdbcUrl.nonEmpty)
+  }
+}
+```
+
+Usage of `TestContainerForEach` is not different from the example above.
+
+#### Multiple containers in tests
+
+If you want to use multiple containers in your test:
+```scala
+class ExampleSpec extends FlatSpec with TestContainersForAll {
```

Comment:
would it be possible to replace abstract type with a type parameter?
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll[MySQLContainer and PostgreSQLContainer]
```
I think this way it would be less verbose

Reply:
At first test traits had type parameters, as you suggest. But I changed it to the type member. There are  3 reasons:
1. You can't refer to a type parameter inside your test body. This is important, because `startContainers` returns` Containers` type, and `withContainers` receives a function with type` Containers => Unit`.
2. If you have a lot of containers it would looks weird in my opinion. I have a service with 6 or 7 containers in testcontainers tests, so this is the real situation.
3. I found that type member usage is a bit more explicit. For example:
    ```scala
    class ExampleSpec extends FlatSpec with TestContainersForAll[MySQLContainer and PostgreSQLContainer] {
      override def startContainers(): Containers = {
        val container1 = MySQLContainer.Def().start ()
        val container2 = PostgreSQLContainer.Def().start ()
        container1 and container2
      }
      
      it should "test" in withContainers { case mysqlContainer and pgContainer =>
        assert(mysqlContainer.jdbcUrl.nonEmpty && pgContainer.jdbcUrl.nonEmpty)
      }
      
    }
    ```
    For me, it looks like `MySQLContainer and PostgreSQLContainer` are lost inside test definition. But here they are a lot more visible and explicit:
    ```scala
    class ExampleSpec extends FlatSpec with TestContainersForAll {

      override type Containers = MySQLContainer and PostgreSQLContainer

      override def startContainers(): Containers = {
        val container1 = MySQLContainer.Def().start ()
        val container2 = PostgreSQLContainer.Def().start ()
        container1 and container2
      }
      
      it should "test" in withContainers { case mysqlContainer and pgContainer =>
        assert(mysqlContainer.jdbcUrl.nonEmpty && pgContainer.jdbcUrl.nonEmpty)
      }
      
    }
    ```